nengo/simulator.py

```diff
@@ -285,10 +291,18 @@ def run(self, time_in_seconds, progress_bar=None):
             For more control over the progress bar, pass in a `.ProgressBar`
             or `.ProgressUpdater` instance.
         """
-        steps = int(np.round(float(time_in_seconds) / self.dt))
-        logger.info("Running %s for %f seconds, or %d steps",
-                    self.model.label, time_in_seconds, steps)
-        self.run_steps(steps, progress_bar=progress_bar)
+        try:
+            steps = int(np.round(float(time_in_seconds) / self.dt))
+        except ZeroDivisionError:
+            steps = 0
+
+        if steps <= 0:
+            warnings.warn("0 timesteps requested. Simulator still at time %g."
+                          % (self.time,))
+        else:
+            logger.info("Running %s for %f seconds, or %d steps",
+                        self.model.label, time_in_seconds, steps)
```

Comment:
Would it make sense to log all requested runs? That might be helpful debug information if someone wonders why the simulator is not advancing as much as it should (though it might be redundant with the warning).

Reply:
You can set up the logger to [log all warnings](https://docs.python.org/3/library/logging.html#logging.captureWarnings), but yeah I don't have a strong preference either way...